### PR TITLE
feat(accordion): Introduce accordion

### LINKED ISSUE #1387: Introduce the Accordion Component
The Accordion component was just merged in Core - https://github.com/patternfly/patternfly-next/pull/1357.

For any CSS questions talk with @christiemolloy

### PR DESCRIPTION
**Pull request by jessiehuff**
**What**: Resolves #1387

**Additional issues**:

### CODE REVIEW

File: packages/patternfly-4/react-core/src/components/Accordion/AccordionToggle.js [hunk 1]
```diff
@@ -3,35 +3,31 @@ import PropTypes from 'prop-types';
 import { css } from '@patternfly/react-styles';
 import { AngleRightIcon } from '@patternfly/react-icons';
 import styles from '@patternfly/patternfly/components/Accordion/accordion.css';
-import { Button } from '@patternfly/react-core';
```
Comment (dlabaj) [inline packages/patternfly-4/react-core/src/components/Accordion/AccordionToggle.js]: Why did we switch this to not use a PF4 button?

Comment (dlabaj) [reply packages/patternfly-4/react-core/src/components/Accordion/AccordionToggle.js]: Okay saw @mcoker comment for this.  Can we delete the commented out Button for pf4?

Comment (jessiehuff) [reply packages/patternfly-4/react-core/src/components/Accordion/AccordionToggle.js]: Absolutely. :)

**Comment by mcoker** [comment]
Not sure if this is already on the list, but we also need to add `.pf-m-expanded` to the expanded `dd`, so that there is a blue line to the left of the button/toggle and expanded content, like this

Screen Shot 2019-04-26 at 8 43 53 AM

**Comment by jgiardino** [comment]
Looks good, @jessiehuff! I just have a couple of minor comments about things to remove.
 
The section headings that are expand/collapse buttons have aria-label="Details". This attribute should not be there. This overrides the actual heading text that's specified, preventing that text from being announced.

Related to this, the `<AccordionToggle>` should not have the prop `aria-label` included. 

The examples include `aria-label="Primary Content Details"` on the `<AccordionContent>` components. We can keep this as a prop on the component, but to match core we should remove this prop in our examples.

File: packages/patternfly-4/react-core/src/components/Accordion/AccordionContent.js [hunk 2]
```diff
@@ -38,7 +40,9 @@ AccordionContent.defaultProps = {
   className: '',
   id: '',
   isHidden: false,
-  isFixed: false
+  isFixed: false,
+  isExpanded: false,
```
Comment (dlabaj) [inline packages/patternfly-4/react-core/src/components/Accordion/AccordionContent.js]: You can remove this you only need isHidden.

File: packages/patternfly-4/react-core/src/components/Accordion/AccordionContent.js [hunk 3]
```diff
@@ -28,8 +28,10 @@ AccordionContent.propTypes = {
   isHidden: PropTypes.bool,
   /** Flag to indicate Accordion content is fixed */
   isFixed: PropTypes.bool,
+  /** Flag to show if the expanded content of the Accordion item is visible */
```
Comment (dlabaj) [inline packages/patternfly-4/react-core/src/components/Accordion/AccordionContent.js]: Not needed on this component.

File: packages/patternfly-4/react-core/src/components/Accordion/AccordionContent.js [hunk 4]
```diff
@@ -3,10 +3,10 @@ import PropTypes from 'prop-types';
 import { css } from '@patternfly/react-styles';
 import styles from '@patternfly/patternfly/components/Accordion/accordion.css';
 
-const AccordionContent = ({ className, children, id, isHidden, isFixed, 'aria-label': ariaLabel, ...props }) => (
+const AccordionContent = ({ className, children, id, isHidden, isFixed, isExpanded, 'aria-label': ariaLabel, ...props }) => (
   <dd
     id={id}
-    className={css(styles.accordionExpandedContent, isFixed && styles.modifiers.fixed, className)}
+    className={css(styles.accordionExpandedContent, isFixed && styles.modifiers.fixed, className, isExpanded && styles.modifiers.expanded)}
```
Comment (dlabaj) [inline packages/patternfly-4/react-core/src/components/Accordion/AccordionContent.js]: className should be last so consumers can override our styles.

**Comment by dlabaj** [comment]
Looks Great Thanks @jessiehuff !

**Comment by aljesusg** [comment]
Error in import https://github.com/patternfly/patternfly-react/issues/1869  we can't upgrade :(

node_modules/@patternfly/react-core/dist/js/components/Accordion/Accordion.d.ts
(2,22): Cannot find module '../../typeUtils'

```typescript
import { Omit } from '../../typeUtils';
```

Fix with 

```typescript
import { Omit } from '../../helpers/typeUtils';
```

**Pull request merged**